### PR TITLE
Add type definitions for @11ty/eleventy-plugin-rss

### PR DESCRIPTION
**Pull request by remcohaszing**
Eleventy doesn’t provide official type definitions, nor do they exist on DefinitelyTyped. However, there is a community project that does this named [`11ty.ts`](https://github.com/panoply/e11ty/tree/master/packages/11ty.ts). These type definitions are based on that.

Docs are based on https://www.11ty.dev/docs/plugins/rss/

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/11ty__eleventy-plugin-rss/11ty__eleventy-plugin-rss-tests.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import pluginRss = require("@11ty/eleventy-plugin-rss");
+import { EleventyConfig } from "11ty.ts";
+import { closingSingleTagOptionEnum } from "posthtml-render";
```
Comment (jakebailey) [inline types/11ty__eleventy-plugin-rss/11ty__eleventy-plugin-rss-tests.ts:3]: This is a dev only dep; do you actually need it? That'd obviate https://github.com/microsoft/DefinitelyTyped-tools/pull/1104

Comment (jakebailey) [reply types/11ty__eleventy-plugin-rss/11ty__eleventy-plugin-rss-tests.ts:3]: Er, no, sorry, my ctrl+f failed me

File: types/11ty__eleventy-plugin-rss/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                name: string;
+                email?: string;
+            };
+        }
+
+        interface Options {
+            /**
+             * @default 'atom'
+             */
+            type?: "atom" | "rss" | "json";
```
Comment (jakebailey) [inline types/11ty__eleventy-plugin-rss/index.d.ts:34]: I highly suspect that these should all have `| undefined` to support `exactOptionalPropertyTypes`.

See: [Playground Link](https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true&ts=4.4.2#code/PTAEHUEsBcAtQKYA8CGBjaB5ADtSB7AOxQBsAFAJ32wQugE8AVemgZ0WICMSEATAGlDMaAZTQVIuALAAoEKFgp2raBIy1QFAK492KKlsK9QAd0XRQ9fFtBoUhUKwQX7oaniKlZ87FRp16UGh8ADpZbzAAMXwKRFQAW2weQThIdkhCaFoAM3QEUF4ENBJ9BHY4FAs4Wny00BQI0F9qWgZTWEg0eDsHTnyifuygk3xHVQyAc1YALlAAcl59AGs5t1i5kkgJ2Gg5xpjQGEdYaxJjQnwLPsOHardOACsi6DCZWQysily0fIBVJwoABEELkdNB2ABvWSgUDyRiwfIoThOQg-NxDFCgABupC0+QoCF8ZQQmXYc1Y9BUCHiexkMLQ+BIMXh1IQmCxtAkhQA-LMAESLChLPmgAA+oD5m220D5AG5ZABfcJyMBQODWKoddjZEoTDhIngCIIIgn1U1wAn5HEkPHsMydeBWGw9RpOCwAAwZTIoLPibI5FC5CHdQVGfX5guFgklWx2IvsxndhkK2QyfHdr0aImceEIeru1rxodASaMILTvBDJkgJBI9Vr+BMoHi+BUoAAUigcWIJLhNIY8H7Gl0ikttQc7sg0llUflgqBcjXBPauod2CDss9IBySIFciQKa95AAJRsIAMpE0Icn1BdaABe97TrHKo1S7D6iixBAoISECIKctCBgAhCEabIDk9RlmQRP12U5SBChDAB9Xh8DKQg5mgZCbmKLRChLZMgPTTMVX-fIdRQPV3WQdAsFwUDSEoFoAmEMoQ3iFAljKIQWAQHtJBcNA0C0ChKgQXdGk46BVzuFMMhAogmiob9CmMJR6gcdxGLrZp-AYUjCmKUo71RDwHAmZx-lobNoFzKYAAoAEpZmsoEQRQMFWHlN4ZAZQg2zdez2AAXlASzoDc2zgucnygsmVgQi9GDWXgwNEPyMKBWWOVZHivNEuSn1YP9BCCKyqU4x8xoACEtAsC5oGmPKcwSpLoOK1KAyDUAwqI1NCD4aqyMoH9izuSjqNojAcHMpi-FaJg+NYd1LzqEwNMappW1YSBuAQEIgA)

Comment (remcohaszing) [reply types/11ty__eleventy-plugin-rss/index.d.ts]: Absolutely! I know and agree this is the right thing to do, but it still doesn’t come to me naturally.

File: types/11ty__eleventy-plugin-rss/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+        interface Metadata {
+            language?: string | undefined;
+            title?: string | undefined;
+            subtitle?: string | undefined;
+            base?: string | undefined;
+            author?: {
+                name: string;
+                email?: string | undefined;
+            };
```
Comment (jakebailey) [inline types/11ty__eleventy-plugin-rss/index.d.ts:27]:
```suggestion
            } | undefined;
```

( easy to miss )

**Comment by jakebailey** [comment]
`11ty.ts` is also missing, it seems.

**Comment by remcohaszing** [comment]
Yep, added in https://github.com/microsoft/DefinitelyTyped-tools/pull/1105.

**Comment by remcohaszing** [comment]
Friendly ping @jakebailey. CI just needs a retrigger :)

**Comment by jakebailey** [comment]
You should be able to close reopen as well

**Review by jakebailey — APPROVED** [review]

**Comment by remcohaszing** [comment]
Ready to merge

**Pull request merged**